File: frontend/src/api/notice.ts
import request from '@/utils/request'
import type { ApiResponse, PageResponse } from './types'

export interface NoticeData {
  id?: number
  title: string
  content?: string
  noticeType: number
  priority: number
  publisher?: string
  publishDate?: string
  expireDate?: string
  targetRole: string
  status: number
  viewCount?: number
  createTime?: string
}

export function getNoticePage(params: any) {
  return request.get<ApiResponse<PageResponse<NoticeData>>>('/notice/page', { params })
}

export function getNoticeList() {
  return request.get<ApiResponse<NoticeData[]>>('/notice/list')
}

export function getAdminNoticeList() {
  return request.get<ApiResponse<NoticeData[]>>('/notice/admin/list')
}

export function createNotice(data: NoticeData) {
  return request.post<ApiResponse>('/notice', data)
}

export function updateNotice(data: NoticeData) {
  return request.put<ApiResponse>('/notice', data)
}

export function publishNotice(id: number, status: number) {
  return request.put<ApiResponse>(`/notice/publish/${id}`, null, { params: { status } })
}

export function getNoticeDetail(id: number) {
  return request.get<ApiResponse<NoticeData>>(`/notice/${id}`)
}

export function deleteNotice(id: number) {
  return request.delete<ApiResponse>(`/notice/${id}`)
}

File: backend/src/main/java/com/hrms/module/notice/entity/Notice.java
package com.hrms.module.notice.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("sys_notice")
public class Notice {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String content;
    /** 类型(1公司公告 2制度通知 3活动通知 4系统通知) */
    private Integer noticeType;
    /** 优先级(0普通 1重要 2紧急) */
    private Integer priority;
    private String publisher;
    private LocalDate publishDate;
    private LocalDate expireDate;
    /** 目标角色(all/admin/employee) */
    private String targetRole;
    /** 状态(0草稿 1已发布 2已撤回) */
    private Integer status;
    private Integer viewCount;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
    @TableLogic
    private Integer isDeleted;
}
